icicle/appUtils/ntt/ntt.cuh

```diff
@@ -212,20 +218,55 @@ __global__ void ntt_template_kernel_shared(
         uint32_t oij = i + j;
         uint32_t k = oij + shift_s;
         S tw = r_twiddles[j * n_twiddles_div];
-
-        E u = s == 0 ? arr_g[offset + oij] : arr[oij];
-        E v = s == 0 ? arr_g[offset + k] : arr[k];
+        // twiddles[threadIdx.x] = r_twiddles[j * n_twiddles_div];
+        
+        // arr[threadIdx.x % groups * elements + threadIdx.x / groups * 2] = arr_g[offset + oij];
+        // arr[threadIdx.x % groups * elements + threadIdx.x / groups * 2 + 1] = arr_g[offset + k];
+        __shared__ E u, v;
```

Comment:
Why do `u` and `v` need to be shared variables here? They can be thread local IMO